processing/src/main/java/io/druid/segment/DimensionHandlerUtils.java

```diff
@@ -263,4 +264,24 @@ public static Float convertObjectToFloat(Object valObj)
       throw new ParseException("Unknown type[%s]", valObj.getClass());
     }
   }
+
+  public static Long getIntegralFromFloatString(String floatStr)
+  {
+    BigDecimal convertedBD;
+    try {
+      convertedBD = new BigDecimal(floatStr);
+    } catch (NumberFormatException nfe) {
+      return null;
+    }
+
+    long asLong;
+    try {
+      asLong = convertedBD.longValueExact();
+    } catch (ArithmeticException ae) {
+      // indicates there was a non-integral part, or the BigDecimal was too big for a long
```

Comment:
Hm, I think comparing a long column against a non-integral value should be a valid operation that matches nothing instead of generating an error.

Imagine a hypothetical use case, where a client program generates a common IN filter with a set of computed floating point values (some values may correspond to integral values, others may not).

This filter gets applied to various numeric columns, some long or float (perhaps the columns contain similar information, but with some of them containing data that is known to be restricted to integral values and stored as long).

I feel like it would be inconvenient to have to strip out non-integral values from the filter in this case when applying it to long columns.

Another case where allowing non-integral values would be useful is schema change; suppose a user previously indexed a column as long, but decided they needed it to be floating point in newer segments. IMO, allowing the non-integral values for longs would also be more useful here than failing, the user could then filter with floating point values on segments from both schema versions.